### ISSUE TITLE
AD0001: Analyzer SonarAnalyzer.Rules.CSharp.CertificateValidationCheck throws NullReferenceException

### ISSUE THREAD

**Issue by TimHess** [SonarSource/sonar-dotnet#4710]
### Description

Using any version of SonarAnalyzer.CSharp NuGet package after 8.22.0.31243 produces this warning:
```
CSC : warning AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.CertificateValidationCheck' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. [\Steeltoe\src\Common\src\Common.Http\Steeltoe.Common.Http.csproj]
```

### Repro steps

Simplified version of the real code, but this is what causes the exception:
```csharp
        public static void RestoreCertificateValidation(RemoteCertificateValidationCallback prevValidator)
        {
                ServicePointManager.ServerCertificateValidationCallback = prevValidator;
        }
```

### Known workarounds

Remove the code above or use 8.22.0.31243 or older

### Related information

* C#/VB.NET Plugins version
* Visual Studio version
* MSBuild / dotnet version: MS Build version 17.0.0-preview-21302-02+018bed83d or 16.10.2+857e5a733
* SonarScanner for .NET version (if used)
* Operating System: Win10 or Azure DevOps Hosted linux build agent

**Comment by csaba-sagi-sonarsource** [comment]
Hello @TimHess

With the repro steps you reported I cannot reproduce the AD0001. Could you please send us a not that much simplified version of the real code? (My assumption is that for a real repro we are missing the argument that is passed into the RestoreCertificateValidation method.) Thanks in advance.

**Comment by TimHess** [comment]
Hi @csaba-sagi-sonarsource 

[This file](https://github.com/SteeltoeOSS/Steeltoe/blob/versions/src/Common/src/Common.Http/HttpClientHelper.cs#L134) -- this branch is currently using the latest version of the Sonar Analyzer package

**Comment by csaba-sagi-sonarsource** [comment]
@TimHess 
Thanks a lot for reporting the issue and for sending the reproducer.
I managed to reproduce the problem and find the cause of the exception. The fix will be part of our next release.
